File: Service/Implement/DanhMucSanPhamService.cs
namespace Service.Implement
{
    public class DanhMucSanPhamService : BaseService<DanhMucSanPham, IDanhMucSanPhamRepository>
    , IDanhMucSanPhamService
    {
        private readonly IDanhMucSanPhamRepository _DanhMucSanPhamRepository;      
        private readonly IWebHostEnvironment _WebHostEnvironment;
        public DanhMucSanPhamService(IDanhMucSanPhamRepository DanhMucSanPhamRepository, IWebHostEnvironment WebHostEnvironment) : base(DanhMucSanPhamRepository, WebHostEnvironment)
        {
            _DanhMucSanPhamRepository = DanhMucSanPhamRepository;
            _WebHostEnvironment = WebHostEnvironment;
        }
    }
}

File: CMS/src/app/shared/DanhMucSanPham.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { DanhMucSanPham } from 'src/app/shared/DanhMucSanPham.model';
import { BaseService } from './Base.service';
@Injectable({
    providedIn: 'root'
})
export class DanhMucSanPhamService extends BaseService{

    DisplayColumns001: string[] = ['STT', 'Name', 'DonGia', 'Save'];

    List: DanhMucSanPham[] | undefined;
    ListFilter: DanhMucSanPham[] | undefined;
    FormData!: DanhMucSanPham;    

    constructor(public httpClient: HttpClient) {
        super(httpClient);
        this.Controller = "DanhMucSanPham";
    }
}

